File: funcscript/parser/syntax/FuncScriptParser.GetExpInParenthesis.cs
using funcscript.block;
namespace funcscript.core
{
    public partial class FuncScriptParser
    {
        static ExpressionBlockResult GetExpInParenthesis(ParseContext context, int index)
        {
            ParseNode parseNode = null;
            ExpressionBlock expBlock = null;
            var i = index;
            i = SkipSpace(context, i).NextIndex;
            var i2 = GetLiteralMatch(context, i, "(").NextIndex;
            if (i == i2)
                return new ExpressionBlockResult(expBlock, parseNode, index);
            i = i2;

            i = SkipSpace(context, i).NextIndex;
            var expressionResult = GetExpression(context, i);
            expBlock = expressionResult.Block;
            var nodeExpression = expressionResult.ParseNode;
            if (expressionResult.NextIndex == i)
                expBlock = null;
            else
                i = expressionResult.NextIndex;
            i = SkipSpace(context, i).NextIndex;
            i2 = GetLiteralMatch(context, i, ")").NextIndex;
            if (i == i2)
            {
                context.SyntaxErrors.Add(new SyntaxErrorData(i, 0, ")' expected"));
                return new ExpressionBlockResult(expBlock, parseNode, index);
            }

            i = i2;
            if (expBlock == null)
                expBlock = new NullExpressionBlock();

            parseNode = new ParseNode(ParseNodeType.ExpressionInBrace, index, i - index, new[] { nodeExpression });
            return new ExpressionBlockResult(expBlock, parseNode, i);
        }
    }
}

File: ts/core/src/parser/syntax/FuncScriptParser.GetExpInParenthesis.ts
import { ParseContext, ParseNode, ParseNodeType, ParseResult, SyntaxErrorData } from "../FuncScriptParser.Main";
import { GetExpression } from "./FuncScriptParser.GetExpression";
import { GetLiteralMatch } from "./FuncScriptParser.GetLiteralMatch";
import { SkipSpace } from "./FuncScriptParser.SkipSpace";

export function GetExpInParenthesis(context: ParseContext, index: number): ParseResult {
    let parseNode: ParseNode | null = null;
    let i = index;
    i = SkipSpace(context, i).NextIndex;
    let i2 = GetLiteralMatch(context, i, "(").NextIndex;
    if (i === i2) {
        return {  ParseNode: parseNode, NextIndex: index };
    }
    i = i2;

    i = SkipSpace(context, i).NextIndex;
    const expressionResult = GetExpression(context, i);
    const nodeExpression = expressionResult.ParseNode;
    if (expressionResult.NextIndex> i) {
        i = expressionResult.NextIndex;
    }
    i = SkipSpace(context, i).NextIndex;
    i2 = GetLiteralMatch(context, i, ")").NextIndex;
    if (i === i2) {
        context.SyntaxErrors.push(new SyntaxErrorData(i, 0, "')' expected"));
        return {  ParseNode: parseNode, NextIndex: index };
    }

    i = i2;

    parseNode = new ParseNode(ParseNodeType.ExpressionInBrace, index, i - index, [nodeExpression!]);
    return {  ParseNode: parseNode, NextIndex: i };
}